### DISCUSSION TITLE
Get table name

### DISCUSSION THREAD

**Discussion by ithilbor** [mikro-orm/mikro-orm#3075] [Q&A] [answered]
Hi i'm using ESM 6 module to write my backend with Mikro-ORM and i was wonder if there's a method to get the table name in order to make this code work:

```
export const initializeORM = async () => {
    const DI = {} as any;
    DI.orm = await MikroORM.init(settings);
    DI.em = DI.orm.em as EntityManager;

    for (const entityInstance of allEntities) {
        if (entityInstance.label === "BaseEntity") {
          continue;
        }
        DI[entityInstance.label] = await DI.orm.em.getRepository(
          entityInstance.entity
        );
      }
      return DI;
};
```

Because even if i import an array with all the modules i can't access to the tableName property declared into the @Entity() decorator. If you know how to get the Repository of all entities and access to that  property let me know how, Thanks in advance.

**Comment by B4nan** [comment, marked as answer]
There is no need for this, repositories are cached in the EM, you dont need to create them upfront, just call `em.getRepository` dynamically. 

Anyway, you can get the metadata via `orm.getMetadata()` and extract whatever information you need. Note that abstract entities will _not_ be there, those are inlined, they can't have a table name, as they don't represent any database table. Same applies for embeddables.

**Reply by ithilbor** [reply]
So in order to get the things work i can change my code into:

```
export const initializeORM = async () => {
    const DI = {} as any;
    DI.orm = await MikroORM.init(settings);
    DI.em = DI.orm.em as EntityManager;
    DI.orm.em.getRepository();

      return DI;
};
```

And this will automatically get all the entities?

**Reply by B4nan** [reply]
Nope, I said you should use `em.getRepository()` directly when you need it. You can't call it without any parameters.

**Reply by ithilbor** [reply]
Perfectly get it, instead of calling it for all the entities in the bootstrap i will call it only when i need it maybe in arrow function that is doing a post request in order to create a new user, right? 
And i need to have a em instance for each different request, in order to have concurrently run different requests? In other words, if I have different posts calls for different entities and different updates calls for different entities, each function needs to have his own em? 
At the end of each request should i call also the em.clear()?
Thanks in advance.
